library/Zend/Db/Sql/Predicate/Expression.php

```diff
@@ -19,7 +19,7 @@ class Expression extends BaseExpression implements PredicateInterface
      * @param string $expression
      * @param int|float|bool|string|array $valueParameter
      */
-    public function __construct($expression = null, $valueParameter = null /*[, $valueParameter, ... ]*/)
+    public function __construct($expression = null, $valueParameter = array() /*[, $valueParameter, ... ]*/)
```

Comment:
I can't see how this fixes your issue.
`array_slice(func_get_args(), 1)` does not care about default values, and returns an empty array no matter if `$valueParameter = null` or `$valueParameter = array()`.